pkgs/macro_service/lib/src/macro_service.g.dart

```diff
@@ -157,6 +157,18 @@ extension type MacroRequest.fromJson(Map<String, Object?> node) {
   }
 }
 
+/// The macro to host protocol version and encoding. TODO(davidmorgan): add the version.
+extension type Protocol.fromJson(Map<String, Object?> node) {
+  Protocol({
+    String? encoding,
```

Comment:
We should make the encoding be an enum (can still be represented as a String on the wire). But the Dart code shouldn't be using strings to create protocols. Might need to update the generator for that but I think it is worth doing.

If we want to leave the door open to more protocols later, we should do the private enum value trick to prevent exhaustive matching.

Could be a TODO though.